packages/permission-controller/src/PermissionController.test.ts

```diff
@@ -2761,7 +2751,6 @@ describe('PermissionController', () => {
       const controller = getDefaultPermissionController();
       const origin = 'metamask.io';
 
-      // TODO: Replace `any` with type
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
       const circular: any = { foo: 'bar' };
```

Comment:
nit: How about removing the `any` altogether and adding `@ts-expect-error` to point out where the invalid inputs are being used?

```diff
diff --git a/packages/permission-controller/src/PermissionController.test.ts b/packages/permission-controller/src/PermissionController.test.ts
index aff18cdf..b3e7ef34 100644
--- a/packages/permission-controller/src/PermissionController.test.ts
+++ b/packages/permission-controller/src/PermissionController.test.ts
@@ -2751,8 +2751,10 @@ describe('PermissionController', () => {
       const controller = getDefaultPermissionController();
       const origin = 'metamask.io';
 
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      const circular: any = { foo: 'bar' };
+      const circular: { foo: string; circular: Record<string, unknown> } = {
+        foo: 'bar',
+        circular: { foo: 'bar' },
+      };
       circular.circular = circular;
 
       [{ foo: () => undefined }, circular, { foo: BigInt(10) }].forEach(
@@ -2765,6 +2767,7 @@ describe('PermissionController', () => {
                   caveats: [
                     {
                       type: CaveatTypes.filterArrayResponse,
+                      // @ts-expect-error Intentional destructive testing
                       value: invalidValue,
                     },
                   ],
```

Reply:
Yep, will do. One note: restating `circular: { foo: 'bar' }` does not produce the desired effect; we want an actual self-referential cycle. I'll add a comment highlighting that.

Reply:
Ah got it. In that case maybe this might be more accurate?

```ts
const circular: {
        foo: 'bar';
        circular?: typeof circular;
      } = {
        foo: 'bar',
      };
```

Reply:
`Record<string, unknown>` seems to do the trick.

Reply:
That would be a bit wider than necessary, but I guess if we want to enforce the recursive property with types down the road, we can always switch to using `typeof circular`. Looks good!